### PR TITLE
Adapt to new `make` pattern + improve tests

### PR DESCRIPTION
**Pull request by davedbase**
@thetarnav before I move ahead with polishing this, can I get some feedback on the primitives API with the new `make` prim? If this makes sense I'll go ahead and polish up the package and adapt the tests if needed.

### CODE REVIEW

**Comment by davedbase** [comment]
- [ ] Finalize API design
- [ ] Make functional minimal demo
- [ ] Port and validate tests

File: packages/intersection-observer/src/index.ts [hunk 1]
```diff
@@ -57,31 +57,97 @@ export type E = JSX.Element;
...
+ * );
+ * ```
+ */
+export const createIntersectionObserver = (
+  elements: MaybeAccessor<Element[]>,
+  onChange: IntersectionObserverCallback,
+  options?: IntersectionObserverInit
+) => {
+  const { add, reset } = makeIntersectionObserver([], onChange, options);
+  createEffect(() => {
```
Comment (thetarnav) [inline packages/intersection-observer/src/index.ts:106]: Effect could be created only if the `elements` arg in an accessor.

File: packages/intersection-observer/README.md [hunk 2]
```diff
@@ -104,4 +105,8 @@ Updated to Solid 1.3
 
 Minor improvements
 
+1.3.0
```
Comment (thetarnav) [inline packages/intersection-observer/README.md:108]: Breaking changes should probably have a major release.

**Review by thetarnav — COMMENTED** [review]
Looks good.
Have you seen this bug: #116 ? Might be worth to check it by the way. It doesn't look like something that we could do on our end, but maybe.

File: packages/intersection-observer/README.md (hunk 2 above)
Comment (davedbase) [reply packages/intersection-observer/README.md:108]: Yup, I'll bump everything when I'm confident it's ready. Sorry just threw this together really fast.

File: packages/intersection-observer/src/index.ts (hunk 1 above)
Comment (davedbase) [reply packages/intersection-observer/src/index.ts:106]: Well if we do that it's just basically a `makeIntersectionObserver` so I've changed it to force an Accessor since that's the only thing that makes sense here. I feel like it's so stripped down now but perhaps this is indeed the basic reactive use?

Comment (thetarnav) [reply packages/intersection-observer/src/index.ts:106]: That would be the point in using it I think.

File: packages/intersection-observer/src/index.ts [hunk 3]
```diff
@@ -57,31 +57,97 @@ export type E = JSX.Element;
...
+export const createVisibilityObserver = (
+  element: MaybeAccessor<Element>,
+  options?: IntersectionObserverInit & {
+    initialValue?: boolean;
+    once?: boolean;
+  }
+): [Accessor<boolean>, { start: () => void; stop: () => void; instance: IntersectionObserver }] => {
+  const [isVisible, setVisible] = createSignal(options?.initialValue ?? false);
+  const { start, stop, instance } = makeIntersectionObserver(
+    [access(element)],
```
Comment (thetarnav) [inline packages/intersection-observer/src/index.ts:138]: I believe this isn't reactive now, if the element in an accessor. It would have to be `createIntersectionObserver` for an accessor, and `makeIntersectionObserver` for just element.

File: packages/intersection-observer/src/index.ts (hunk 1 above)
Comment (davedbase) [reply packages/intersection-observer/src/index.ts:106]: Yes yes, lol. I agree.

File: packages/intersection-observer/src/index.ts (hunk 3 above)
Comment (davedbase) [reply packages/intersection-observer/src/index.ts]: Oops good point. I added an effect if an accessor is supplied.

**Comment by davedbase** [comment]
> Looks good. Have you seen this bug: #116 ? Might be worth to check it by the way. It doesn't look like something that we could do on our end, but maybe.

Yeah I don't think we can handle that edge case without doing some JS acrobatics. Not worth the effort IMHO.

**Comment by davedbase** [comment]
@thetarnav I think I'm going to call this a wrap and implement the tests + demo.

**Comment by davedbase** [comment]
@thetarnav @atk ok this is ready for final-final review.

**Review by atk — APPROVED** [review]
